File: pat1048.py
"""
    注：测试点3运行超时
"""


N, M = [int(x) for x in input().split()]
coins = [int(x) for x in input().split()]

coins.sort()
has_solution = False
all_values = [0] * 1000
for coin_value in coins:
    all_values[coin_value] += 1

for coin_value in coins:
    num = M - coin_value
    if coin_value == num and all_values[num] == 1:
        continue
    if all_values[num] >= 1:
        print(str(coin_value) + " " + str(num))
        has_solution = True
        break

if not has_solution:
    print("No Solution")

File: pat1048.cpp
#include<iostream>
#include<algorithm>
using namespace std;

int coins[100010], coin_count[1000];	// 记录输入值，以及出现硬币的次数
int main() {

	int N, M;
	cin >> N >> M;
	for (int i = 0; i < N; i++) {
		cin >> coins[i];
		coin_count[coins[i]] += 1;
	}

	sort(coins, coins + N);
	bool has_solution = false;
	for (int i = 0; i < N; i++) {
		int num = M - coins[i];
		if (num == coins[i] && coin_count[num] == 1) {
			continue;
		}
		if (coin_count[num] >= 1) {
			cout << coins[i] << " " << num;
			has_solution = true;
			break;
		}
	}

	if (!has_solution) {
		cout << "No Solution";
	}


	return 0;
}